error TS2665: Invalid module name in augmentation in jest/async-storage-mock

### What happened?

Just upgraded to 1.16.0, my typescript builds started to error with:
```
node_modules/@react-native-async-storage/async-storage/lib/typescript/index.d.ts:6:16 - error TS2665: Invalid module name in augmentation. Module '@react-native-async-storage/async-storage/jest/async-storage-mock' resolves to an untyped module at '[...]/node_modules/@react-native-async-storage/async-storage/jest/async-storage-mock.js', which cannot be augmented.

6 declare module '@react-native-async-storage/async-storage/jest/async-storage-mock' {
```

### Version

1.16.0

### What platforms are you seeing this issue on?

- [X] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows
- [ ] web

### System Information

```true
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 AMD Ryzen 7 3700U with Radeon Vega Mobile Gfx
    Memory: 6.79 GB / 17.88 GB
  Binaries:
    Node: 14.17.1 - ~\AppData\Local\Temp\yarn--1644419667475-0.14216515503697358\node.CMD
    Yarn: 1.22.5 - ~\AppData\Local\Temp\yarn--1644419667475-0.14216515503697358\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7935034
    Visual Studio: 16.11.32002.261 (Visual Studio Community 2019)
  Languages:
    Java: 1.8.0_282 - C:\Program Files\AdoptOpenJDK\jdk-8.0.282.8-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.66.4 => 0.66.4
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```


### Steps to Reproduce

yarn install "@react-native-async-storage/async-storage@1.16.0
yarn tsc

I just noticed that I don't have the issue that the author has. **BUT** the issue that the author has seems to come from trying to solves this.

I'm getting:

> error TS7016: Could not find a declaration file for module '@react-native-async-storage/async-storage/jest/async-storage-mock'. '/var/app/node_modules/@react-native-async-storage/async-storage/jest/async-storage-mock.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/react-native-async-storage__async-storage` if it exists or add a new declaration (.d.ts) file containing `declare module '@react-native-async-storage/async-storage/jest/async-storage-mock';`

If I then add `declare module '@react-native-async-storage/async-storage/jest/async-storage-mock';` in my code, I get: 

> error TS2665: Invalid module name in augmentation. Module '@react-native-async-storage/async-storage/jest/async-storage-mock' resolves to an untyped module at '/var/app/node_modules/@react-native-async-storage/async-storage/jest/async-storage-mock.js', which cannot be augmented.

So... the original issue probably isn't the root cause.

It's also really weird, I found the initial import to sometimes work, and suddenly stop (in new project). Or it could be a weird caching thing, from VSCode or whatever. But the following docker container reproduces **my** issue, every tie (for me).

Ah, and it is noteworthy to say, that I do not put the mock into the jest setup file, like stated here: https://react-native-async-storage.github.io/async-storage/docs/advanced/jest/#with-jest-setup-file. I just import it in the tests that require the mock.

## Instructions

Create a random directory and place the following files inside it:

### `index.ts` 

```ts
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

function test() {
  mockAsyncStorage();
}
```

### `tsconfig.json`

```json
{
  "compilerOptions": {
    "target": "es2016",
    "module": "commonjs",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

### `package.json`

```json
{
  "name": "react-native-async-storage-mock-test",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@react-native-async-storage/async-storage": "^1.16.1",
    "typescript": "^4.5.5"
  },
  "description": ""
}
```

### `Dockerfile`

```dockerfile
FROM node:16-alpine

WORKDIR /var/app

COPY package.json tsconfig.json index.ts ./

RUN npm install

CMD ["npx", "tsc", "--noEmit"]
```

Now just build and run that container. (inside the created directory: `docker build .`, after that ran there should be an id at the bottom (successfully built <id>), `docker run <id by the previous command>`).

--- a/jest/async-storage-mock.d.ts
+++ b/jest/async-storage-mock.d.ts
@@ -1,0 +1,9 @@
+import type {
+  AsyncStorageHook,
+  AsyncStorageStatic,
+} from '../lib/typescript/types';
+
+export function useAsyncStorage(key: string): AsyncStorageHook;
+
+declare const AsyncStorage: AsyncStorageStatic;
+export default AsyncStorage;

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,14 +1,5 @@
 import AsyncStorage from './AsyncStorage';
-import type { AsyncStorageHook, AsyncStorageStatic } from './types';
 
 export { useAsyncStorage } from './hooks';
-export type { AsyncStorageStatic } from './types';
 
 export default AsyncStorage;
-
-// @ts-ignore AsyncStorage mock module
-declare module '@react-native-async-storage/async-storage/jest/async-storage-mock' {
-  export function useAsyncStorage(key: string): AsyncStorageHook;
-  const AsyncStorageLib: AsyncStorageStatic;
-  export default AsyncStorageLib;
-}

